docs/resources/server/recipes/ingredients.md

```diff
@@ -1,143 +1,184 @@
 
-```json5
-// For some input
-{
-    "type": "neoforge:components",
-    // Can be either a single item or list of items
-    "items": "examplemod:example_item",
-    "components": {
-        // Add component information
-    },
-    // true requires an exact match
-    // false only requires a match of the specified components
-    "strict": true
-}
-```
+```java
+public class MinEnchantedIngredient implements ICustomIngredient {
+    private final TagKey<Item> tag;
+    private final Map<Holder<Enchantment>, Integer> enchantments;
+    // The codec for serializing the ingredient.
+    public static final MapCodec<MinEnchantedIngredient> CODEC = RecordCodecBuilder.mapCodec(inst -> inst.group(
+            TagKey.codec(Registries.ITEM).fieldOf("tag").forGetter(e -> e.tag),
+            Codec.unboundedMap(Enchantment.CODEC, Codec.INT)
+                    .optionalFieldOf("enchantments", Map.of())
+                    .forGetter(e -> e.enchantments)
+    ).apply(inst, MinEnchantedIngredient::new));
+    // Create a stream codec from the regular codec. In some cases, it might make sense to define
+    // a new stream codec from scratch.
+    public static final StreamCodec<RegistryFriendlyByteBuf, MinEnchantedIngredient> STREAM_CODEC =
+            ByteBufCodecs.fromCodecWithRegistries(CODEC.codec());
+
+    // Allow passing in a pre-existing map of enchantments to levels.
+    public MinEnchantedIngredient(TagKey<Item> tag, Map<Holder<Enchantment>, Integer> enchantments) {
+        this.tag = tag;
+        this.enchantments = enchantments;
+    }
+
+    public MinEnchantedIngredient(TagKey<Item> tag) {
+        this(tag, new HashMap<>());
+    }
 
-### IntersectionIngredient
+    // Make this chainable for easy use.
+    public MinEnchantedIngredient with(Holder<Enchantment> enchantment, int level) {
+        enchantments.put(enchantment, level);
+        return this;
+    }
 
-`IntersectionIngredient`s work as a set AND where the passed in stack must match all supplied ingredients. There must be at least two ingredients supplied to this. This can be used by specifying the `type` as `neoforge:intersection`.
+    // Check if the passed ItemStack matches our ingredient by verifying the item is in the tag
+    // and by testing for presence of all required enchantments with at least the required level.
+    @Override
+    public boolean test(ItemStack stack) {
+        return stack.is(tag) && enchantments.keySet()
+                .stream()
+                .allMatch(ench -> stack.getEnchantmentLevel(ench) >= enchantments.get(ench));
+    }
 
-```json5
-// For some input
-{
-    "type": "neoforge:intersection",
-
-    // All of these ingredients must return true to succeed
-    "children": [
-        {
-            // Ingredient 1
-        },
-        {
-            // Ingredient 2
+    // Determines whether this ingredient performs NBT or data component matching (true) or not (false).
```

Comment:
The return value has exactly the opposite meaning (the javadoc on the method is easy to misread). It should therefore also return false in this example.